__react/template/src/index.ts.js

```diff
@@ -0,0 +1,146 @@
+import { Publish } from '../../components/index/publish';
+import { Subscribe } from '../../components/index/subscribe';
+import { Reply } from '../../components/index/reply';
+import { Request } from '../../components/index/request';
+import { camelCase, pascalCase, isRequestReply, isReplier, isRequester, isPubsub} from '../../utils/index';
+
+// eslint-disable-next-line sonarjs/cognitive-complexity
+/**
+ * Return the correct channel functions for the client based on whether a channel is `pubSub` or `requestReply`.
+ * 
+ * @param {*} asyncapi 
+ * @param {*} params 
+ */
+function getChannelWrappers(asyncapi, params) {
+  let channelWrappers = [];
+  channelWrappers = Object.keys(asyncapi.channels()).length ? Object.entries(asyncapi.channels()).map(([channelName, channel]) => {
+    if (isRequestReply(channel)) {
+      if (isRequester(channel)) {
+        return Request(
+          asyncapi.defaultContentType(), 
+          channelName, 
+          channel.subscribe().message(0),
+          channel.publish().message(0),
+          channel.description(),
+          channel.parameters()
+        );
+      }
+      if (isReplier(channel)) {
+        return Reply(
+          asyncapi.defaultContentType(), 
+          channelName, 
+          channel.subscribe().message(0),
+          channel.publish().message(0),
+          channel.description(),
+          channel.parameters(),
+          params
+        );
+      }
+    }
+
+    if (isPubsub(channel)) {
+      if (channel.hasSubscribe()) {
+        return Publish(
+          asyncapi.defaultContentType(), 
+          channelName, 
+          channel.subscribe().message(0), 
+          channel.description(), 
+          channel.parameters());
+      }
+      if (channel.hasPublish()) {
+        return Subscribe(
+          asyncapi.defaultContentType(), 
+          channelName, 
+          channel.publish().message(0), 
+          channel.description(), 
+          channel.parameters());
+      }
+    }
+  }) : '';
+  return channelWrappers;
+}
+
+
+export default function index({ asyncapi, params }) {
+  //Import the channel code and re-export them
+  let channelImport = asyncapi.channels();
+  channelImport = Object.keys(channelImport).length ? Object.entries(channelImport).map(([channelName, _]) => {
+    return `
+      import * as ${camelCase(channelName)}Channel from "./channels/${pascalCase(channelName)}";
+      export {${camelCase(channelName)}Channel};
+    `;
+  }) : '';
+
+  //Import the messages and re-export them
+  const messagesImport = [];
+  for (const [messageName] of asyncapi.allMessages()) {
+    const pascalMessageName = pascalCase(messageName);
+    messagesImport.push(`
+      import * as ${pascalMessageName}Message from "./messages/${pascalMessageName}";
+      export {${pascalMessageName}Message};
+    `);
+  }
+
+  return (
+    <File name="index.ts">
+      {
+        `
+        import {fromSeed} from 'ts-nkeys';
```

Comment:
I know about issues with indentation, but remember, everything in this file will be intend :)

Reply:
Also split imports and exports to seperate groups in file:

```ts
import ...
import ...

export ...
export...
```

Reply:
I like the idea 😄